moneyed/classes.py

```diff
@@ -124,37 +152,42 @@ class Money:
     ($DEITY forbid) floats.
     """
 
-    def __init__(self, amount=Decimal("0.0"), currency=DEFAULT_CURRENCY_CODE):
-        if not isinstance(amount, Decimal):
-            amount = Decimal(str(amount))
-        self.amount = amount
-
-        if not isinstance(currency, Currency):
-            currency = get_currency(str(currency).upper())
-        self.currency = currency
+    def __init__(
+        self,
+        amount: object = Decimal("0.0"),
+        currency: Union[str, Currency] = DEFAULT_CURRENCY_CODE,
+    ) -> None:
+        self.amount: Final = (
```

Comment:
Does it make sense to specify the type in square brackets - `Final[Decimal]`, or is it inferred by mypy?

Reply:
It's inferred. I double-checked now by adding a `reveal_type(self.amount)` below which outputs: `moneyed/classes.py:163: note: Revealed type is 'decimal.Decimal'`. I usually don't add the type in these situations because it's already explicitly defined in the `__init__` signature. When adding `Final` to standalone constants I usually add the type within square brackets though :)

Reply:
:+1: